src/main/java/org/haffson/adventofcode/days/day01/Day01.java

```diff
@@ -4,29 +4,36 @@
 import org.haffson.adventofcode.days.Days;
 import org.haffson.adventofcode.utils.FileReaders;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
+import java.util.Arrays;
 import java.util.HashMap;
 
 /**
- * Implementation for <i>Day 1: Chronal Calibration</i>.
+ * Implementation for <i>Day 1: Report Repair/i>.
  */
 @Component
 public class Day01 implements Days {
 
     /** The puzzle status {@code HashMap} */
     private final HashMap<String, ProblemStatusEnum> problemStatus;
 
-    /**
-     * Causes the input file to be parsed into the frequencies array ({@code frequencies}).
-     *
-     * @param fileReaders {@code @Autowired} fileReader //TODO: inject what you need
-     */
+    private final int[] numbers;
+
+    Day01() {
+        this.problemStatus = new HashMap<>();
+        this.problemStatus.put("1", ProblemStatusEnum.SOLVED);
+        this.problemStatus.put("2", ProblemStatusEnum.SOLVED);
+        this.numbers = new int[]{};
+    }
+
     @Autowired
-    Day01(FileReaders fileReaders) {
+    Day01(FileReaders fileReaders, @Value("${day1.file}") String filePath) {
         this.problemStatus = new HashMap<>();
-        this.problemStatus.put("1", ProblemStatusEnum.UNSOLVED);
-        this.problemStatus.put("2", ProblemStatusEnum.UNSOLVED);
+        this.problemStatus.put("1", ProblemStatusEnum.SOLVED);
+        this.problemStatus.put("2", ProblemStatusEnum.SOLVED);
+        this.numbers = fileReaders.readFileIntoArrayOfIntegers(filePath);
```

Comment:
You may want to delay reading the file until the numbers are needed.

In this case, the advantage would be an improved startup-time. The disadvantage would be, that you cannot validate the file contents immediately.

Another point: Simple constructors usually simplify testing.

Reply:
The reading of the puzzle values is not optimal in the constructor. I am still looking at how to solve it differently.